Python/aot_ceval_jit.c

```diff
@@ -847,7 +900,11 @@ static void switch_section(Jit* Dst, Section new_section) {
 // compares r_type_idx->tp_version_tag with type_ver
 // branches to false_branch on inequality else continues
 |.macro type_version_check, r_type_idx, type_ver, false_branch
-|| emit_cmp64_mem_imm(Dst, r_type_idx, offsetof(PyTypeObject, tp_version_tag), (uint64_t)type_ver);
+#ifdef PYSTON_LITE
+|| emit_cmp32_mem_imm(Dst, r_type_idx, offsetof(PyTypeObject, tp_version_tag), (unsigned int)type_ver);
```

Comment:
I had totally forgot about the difference in size of `tp_version_tag`, good that you noticed!
But the #ifdef/else/endif needs "||" infront. Right now dynasm generates both paths because it just seems to ignore them:
```C
    } else {
        // PyTypeObject *arg2 = Py_TYPE(obj)
        emit_load64_mem(Dst, arg2_idx, arg1_idx,  offsetof(PyObject, ob_type));
        //| type_version_check, arg2_idx, la->type_ver, >1
        emit_cmp32_mem_imm(Dst, arg2_idx, offsetof(PyTypeObject, tp_version_tag), (unsigned int)la->type_ver);
        emit_cmp64_mem_imm(Dst, arg2_idx, offsetof(PyTypeObject, tp_version_tag), (unsigned int)la->type_ver);
        dasm_put(Dst, 323);
```